scalding_typed/src/main/scala/com/matthewrathbone/scalding/Main.scala

```diff
@@ -0,0 +1,34 @@
+package main.scala.com.matthewrathbone.scalding
+
+import com.twitter.scalding._
+import TDsl._
+import cascading.tuple.TupleEntry
+
+
+class Main ( args: Args ) extends Job( args ) {
+
+	val output = args( "output" )
+	
+	val input1 : TypedPipe[String] = TypedPipe.from(TextLine(args( "input1" )))
+	val input2 : TypedPipe[String] = TypedPipe.from(TextLine(args( "input2" )))
+	
+	val usersInput : TypedPipe[(String, String, String, String)] = input1.map{ s:String =>
+      val split = s.split("\t")
+      (split( 0 ), split( 1 ), split( 2 ), split( 3 ))
+	}
+	val group1 = usersInput.groupBy { case (id, email, language, location) => id }
+
+	val transactionsInput : TypedPipe[(String, String, String, String, String)] = input2.map{ s:String =>
```

Comment:
From the first glance it does not look all that useful, just more lines of code: 

{% highlight scala %}	
case class Transaction(id: Long, productId: Long, userId: Long, purchaseAmount: Double, itemDescription: String){}
	val transactionsInput : TypedPipe[Transaction] = input2.map{ s:String =>
      val split = s.split("\t")
      Transaction(split( 0 ).toLong, split( 1 ).toLong, split( 2 ).toLong, split( 3 ).toDouble, split( 4 ))
    }

	val transactionsFields : TypedPipe[(Long, Long, Long, Double, String)] = transactionsInput.map { t => (t.id, t.productId, t.userId, t.purchaseAmount, t.itemDescription) }
	val group2 = transactionsFields.groupBy{ case (id, productId, userId, purchaseAmount, itemDescription) => userId }

{% endhighlight %}